src/test/lambda/commands/copyLambdaUrl.test.ts

```diff
@@ -0,0 +1,81 @@
+import { DefaultLambdaClient, LambdaClient } from '../../../shared/clients/lambdaClient'
+import globals from '../../../shared/extensionGlobals'
+import { addCodiconToString } from '../../../shared/utilities/textUtilities'
+import { buildFunctionUrlConfig } from '../../shared/clients/defaultLambdaClient.test'
+
+describe('copy lambda function URL to clipboard', async () => {
+    let client: SinonStubbedInstance<LambdaClient>
+    let node: Pick<LambdaFunctionNode, 'name' | 'regionCode'>
+    let clipboardFunc: SinonStub
+    let quickPickUrlFunc: SinonStub
+
+    const nodeName = 'my-node-name'
+
+    beforeEach(async () => {
+        client = createStubInstance(DefaultLambdaClient)
+        clipboardFunc = stub()
+        quickPickUrlFunc = stub()
+
+        node = { name: nodeName, regionCode: 'myRegion' }
+    })
+
+    it('Single URL exists', async () => {
+        const urlConfig = buildFunctionUrlConfig({ FunctionUrl: 'url1', FunctionArn: 'arn1' })
+        client.getFunctionUrlConfigs.resolves([urlConfig])
+
+        await copyLambdaUrl(node, client, clipboardFunc)
+
+        assert(client.getFunctionUrlConfigs.calledOnceWith(nodeName))
+        assert(clipboardFunc.calledOnceWith(urlConfig.FunctionUrl, 'URL'))
+    })
+
+    it('Multiple URLs exists', async () => {
+        const expectedConfig = buildFunctionUrlConfig({ FunctionUrl: 'url1', FunctionArn: 'arn1' })
+        const urlConfigs = [expectedConfig, buildFunctionUrlConfig({ FunctionUrl: 'url2', FunctionArn: 'arn2' })]
+        client.getFunctionUrlConfigs.resolves(urlConfigs)
+        quickPickUrlFunc.resolves(expectedConfig.FunctionUrl)
+
+        await copyLambdaUrl(node, client, clipboardFunc, quickPickUrlFunc)
+
+        assert(client.getFunctionUrlConfigs.calledOnceWith(nodeName))
+        assert.deepStrictEqual(quickPickUrlFunc.args, [[urlConfigs]])
+        assert.deepStrictEqual(clipboardFunc.args, [[expectedConfig.FunctionUrl, 'URL']])
+    })
+
+    describe("URL doesn't exist", async () => {
+        let spiedInformationMessage: SinonSpy
+        let spiedStatusBarMessage: SinonSpy
+
+        before(async () => {
+            spiedInformationMessage = spy(globals.window, 'showWarningMessage')
+            spiedStatusBarMessage = spy(globals.window, 'setStatusBarMessage')
+        })
+
+        afterEach(async () => {
+            spiedInformationMessage.resetHistory()
+            spiedStatusBarMessage.resetHistory()
+        })
+
+        it(`URL doesn't exist`, async () => {
+            client.getFunctionUrlConfigs.resolves([])
+
+            await copyLambdaUrl(node, client, clipboardFunc)
+
+            assert.deepStrictEqual(client.getFunctionUrlConfigs.args, [[nodeName]])
+            assert.deepStrictEqual(clipboardFunc.args, [])
+            assert.deepStrictEqual(spiedInformationMessage.args, [[noLambdaFuncMessage]])
+            assert.deepStrictEqual(spiedStatusBarMessage.args, [
```

Comment:
check actual clipboard. then we can eliminate these 4 spies and possibly some/all of the stubs.

Reply:
I've removed the spy for the clipboard func.
I don't see a way to test the bar+info message without spying, did you have something in mind?

Reply:
no. looks better now!